File: binary search/find_k_closest_elements.java
class Solution {
    public List<Integer> findClosestElements(int[] arr, int k, int x) {
        int l = 0;
        int h = arr.length-1;
        
        while(h-l >= k)
        {
            if (Math.abs(arr[l]-x) > Math.abs(arr[h]-x))
                l++;
            else
                h--;
        }
        
        List<Integer> ans = new ArrayList<>(k);
        for(int i=l; i<=h; i++)
        {
            ans.add(arr[i]);
        }
        
        return ans;
    }
}

File: binary search/find_k_closest_elements.py
class Solution:
    def findClosestElements(self, arr: List[int], k: int, x: int) -> List[int]:
        l = 0
        h = len(arr) - 1
        
        while h-l >= k:
            if abs(arr[l] - x) > abs(arr[h] - x):
                l += 1
            else:
                h -= 1
        
        return arr[l:h+1]